pkg/envoy/rds/route/route_config.go

```diff
@@ -286,6 +287,40 @@ func buildWeightedCluster(weightedClusters mapset.Set) *xds_route.WeightedCluste
 	sort.Stable(clusterWeightByName(wc.Clusters))
 	return &wc
 }
+func buildRetryPolicy(retry *v1alpha1.RetryPolicySpec) *xds_route.RetryPolicy {
+	rp := &xds_route.RetryPolicy{}
+	if retry == nil {
+		return rp
+	}
+	if retry.RetryOn != "" {
+		rp.RetryOn = retry.RetryOn
+	} else {
+		log.Error().Msgf("Retry policy must specify which policies to retry on")
+		return rp
+	}
+	if retry.NumRetries != 0 {
+		rp.NumRetries = &wrapperspb.UInt32Value{
+			Value: uint32(retry.NumRetries),
+		}
+	}
+	if retry.PerTryTimeout != 0 {
+		rp.PerTryTimeout = convertIntToDuration(retry.PerTryTimeout)
+	}
+	if retry.RetryBackoffInterval != 0 {
+		rp.RetryBackOff = &xds_route.RetryPolicy_RetryBackOff{
+			BaseInterval: convertIntToDuration(retry.RetryBackoffInterval),
+		}
+	}
+	return rp
+}
```

Comment:
There are a lot of validations here that should be detected via a validation webhook. I'd suggest directly setting the XDS Retry object based on the values filled in the Retry object in the routes already. This will simplify the error scenarios and make this code simpler.